pkg/sql/colexec/execgen/template.go

```diff
@@ -291,27 +285,57 @@ func createTemplateFuncVariants(f *dst.File) map[string]*funcInfo {
 	return ret
 }
 
-func getTemplateVariantName(info *funcInfo, args []dst.Expr) *dst.Ident {
+// extractTemplateVarsFromComment returns a map from the template variables
+// in an execgen:template declaration to their position within the list.
+func extractTemplateVarsFromComment(n dst.Node) (templateVars map[string]int) {
+	decs := n.Decorations()
+	if len(decs.Start) == 0 {
+		return nil
+	}
+
+	ret := make(map[string]int)
+
+	var templateDecPosition int
+	for i, dec := range decs.Start.All() {
+		if matches := templateRe.FindStringSubmatch(dec); matches != nil {
+			match := matches[1]
+			// Match now looks like foo, bar
+			templateVars := strings.Split(match, ",")
+			for i, v := range templateVars {
+				ret[strings.TrimSpace(v)] = i
+			}
+			templateDecPosition = i
+			break
+		}
+	}
+	// Remove the template decoration.
+	decs.Start = append(
+		decs.Start[:templateDecPosition],
+		decs.Start[templateDecPosition+1:]...)
+	return ret
+}
+
+func getTemplateVariantIdent(objectName string, args []dst.Expr) *dst.Ident {
 	var newName strings.Builder
-	newName.WriteString(info.decl.Name.Name)
+	newName.WriteString(objectName)
```

Comment:
this object name might need some mangling if it contains characters that couldn't be part of a name, like a * or something.